File: src/main/java/by/devpav/serfor/repository/DirectoryRepository.java
package by.devpav.serfor.repository;

import by.devpav.serfor.domain.VirtualDirectory;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface DirectoryRepository extends BasicEntityRepository<VirtualDirectory> {

    VirtualDirectory findByName(String name);

    List<VirtualDirectory> findByRealm_Name(String realmName);

    VirtualDirectory findByRealm_NameAndWidthAndHeight(String realm, Integer width, Integer height);

}

File: src/main/kotlin/by/devpav/serfor/repository/DirectoryRepository.kt
package by.devpav.serfor.repository

import by.devpav.serfor.domain.VirtualDirectory
import org.springframework.stereotype.Repository

@Repository
interface DirectoryRepository : BasicEntityRepository<VirtualDirectory> {

    fun findByName(name: String): VirtualDirectory

    fun findByRealm_Name(realmName: String): List<VirtualDirectory>

    fun findByRealm_NameAndWidthAndHeight(realm: String, width: Int?, height: Int?): VirtualDirectory

}
